docs/source/algorithms/FitPeaks-v1.rst

```diff
@@ -0,0 +1,434 @@
+user's input.
+The first peak will be fit with the starting value provided by the user.
+except background and peak center, which will be determiend by *observation*.
+
+Background
+##########
+
+Linear background is used for fitting peaks.  The background parameters
+will be estimated via *observation*.
+
+Estimation of values of peak profiles
+#####################################
+
+Peak intensity and peak center are estimated by observing the maximum value within peak fit window.
+
+Subalgorithms used
+##################
+
+FitPeaks uses the :ref:`algm-Fit` algorithm to fit each single peak.
+FitPeaks uses the :ref:`algm-FindPeakBackground` algorithm to estimate the background of each peak.
+
+
+Inputs
+======
+
+The inputs tends to be general enough for various use cases.
+
+
+Limitations on Partial Spectra
+##############################
+
+ * For partial spectra peak fitting, the spectra must be consecutive.
+
+
+Peak positions
+##############
+
+One of only one of the following will be taken.
+
+ * A MatrixWorkspace
+   * Number of spectra shall be the same as the number of spectra of the workspace containing peaks to fit for.  Or the number of spectra is the same as the number of spectra of the input workspace.
+   * X value is the index of the peak.
+   * Y value is the position of the peaks to fit.
+   * Peak centers are stored in **m_peakCenterWorkspace**.
+   - Spectra can be a subset of all spectra because FitPeaks can work on partial spectra.
+
+ * An array of double as the positions of the peaks to fit.
+
+   * Peak centers are stored in **m_peakCenters**
+
+
+ **Peaks' positions must be given in ascending order**
+
+ Parameter **FitFromRight** is used as the flag to fit from right most peak or left most peak.
+
+
+
+Fit Window
+##########
+
+There are two input parameters that are associated with fitting window.
+
+ * FitWindowBoundaryList
+ * FitPeakWindowWorkspace
+
+
+If FitWindows is defined, then a peak's range to fit (i.e., x-min and
+x-max) is confined by this window.
+
+If FitWindows is defined, starting peak centres are NOT user's input,
+but found by highest value within peak window. (Is this correct???)
+
+
+Further down the road, here are the fitting setup that can be affected.
+
+  * Peak positions are uniform among all spectra;
+
+    - Peak window information will be retrieved from **m_peakWindowVector**
+
+  * Peak positions are different among spectra.
+
+    - Peak windown information will be retrieved from **m_peakWindowWorkspace**
+
+
+Tolerance on Fitting Peaks Positions
+####################################
+
+Tolerance will be always checked!
+
+ * Uniform tolerance
+ 
+ * Non-uniform tolerance
+
+ * Case 2, 3 and 4
+
+
+
+Algorithm Configurations
+########################
+
+ * Peak profile starting value will be given as 
+
+    - an array **PeakParameterValues** such that the starting values are uniform among all spectra.
```

Comment:
Use back-ticks rather than stars to denote parameters

Reply:
Applied